File: 872. Leaf-Similar Trees.java
class Solution {
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        List<Integer> list1 = new ArrayList();
        List<Integer> list2 = new ArrayList();
        leafNode(root1,list1);
        leafNode(root2,list2);
        return list1.equals(list2);
    }
    public void leafNode(TreeNode root,List list){
      if(root==null)return;
      if(root.left==null&&root.right==null){
          list.add(root.val);
      }
       leafNode(root.left,list);
       leafNode(root.right,list);
    }
 }

File: 872. Leaf-Similar Trees.cpp
class Solution {
    vector<int> x;
    vector<int> y;
public:
    void printLeaf(TreeNode* root, vector<int> &k) {
        if(root == NULL) return;
        printLeaf(root->left, k);
        if(root->left == NULL && root->right == NULL) k.push_back(root->val);
        printLeaf(root->right, k);
    }
    bool leafSimilar(TreeNode* root1, TreeNode* root2) {
        printLeaf(root1, x);
        printLeaf(root2, y);
        return x==y;
    }
};
